valhalla/sif/dynamiccost.h

```diff
@@ -497,10 +497,10 @@ class DynamicCost {
 
     // If forward, check if the edge marks the end of a restriction, else check
     // if the edge marks the start of a complex restriction.
-    if ((forward && (edge->end_restriction() & access_mode())) ||
-        (!forward && (edge->start_restriction() & access_mode()))) {
+    if ((forward && (edge->end_restriction() & access_mask_)) ||
+        (!forward && (edge->start_restriction() & access_mask_))) {
```

Comment:
everything else uses `access_mask_` instead of calling the public api and makes sense as long as we're inside the class